storage/plugins/ncip-server/NcipServerPlugin.php

```
@@ -0,0 +1,1453 @@
+            return null;
+        }
+        $user = $res->fetch_assoc();
+        $stmt->close();
+
+        if (!is_array($user)) { return null; }
+        if (!password_verify($password, (string) ($user['password'] ?? ''))) {
+            return null;
+        }
+
+        return $user;
+    }
```

PHPMD (2.15.0)

[warning] The method authenticate() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)